saffron/src/folding.rs

```diff
@@ -51,22 +51,22 @@ use rand::{CryptoRng, RngCore};
 /// Non-relaxed instance attesting to `d * q - a = 0`
 pub struct CoreInstance {
     /// Commitment to the data
-    comm_d: Curve,
+    pub comm_d: Curve,
```

Comment:
Same than below. I would be in favor of not making the fields public, and keep a clean user API.

Reply:
:thinking: How is creating a method with the same name "cleaner" in terms of public API? I guess what you want to avoid is to minimise API changes, but if we want to change the name of the field we would also want to change the name of the method, so it's the same... the `pub/pub(crate)` accessors are the same for fields and methods as far as I know.

Taking this example to maximum: we could follow java style and wrap every field into `set_field()` and `get_field()`, together with creating a `constructor()`, but I really don't see what it achieves in terms of maintainability (while it adds _a lot_ of visual noise).